In [ ]:
import pandas as pd

In [3]:
# import data 
data = pd.read_csv("2) Stock Prices Data Set.csv", encoding='unicode_escape')

In [5]:
data.shape

(497472, 7)

In [6]:
data.head(5)

,symbol,date,open,high,low,close,volume
0,AAL,2014-01-02,25.0700,25.8200,25.0600,25.3600,8998943
1,AAPL,2014-01-02,79.3828,79.5756,78.8601,79.0185,58791957
2,AAP,2014-01-02,110.3600,111.8800,109.2900,109.7400,542711
3,ABBV,2014-01-02,52.1200,52.3300,51.5200,51.9800,4569061
4,ABC,2014-01-02,70.1100,70.2300,69.4800,69.8900,1148391


In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 497472 entries, 0 to 497471
Data columns (total 7 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   symbol  497472 non-null  object 
 1   date    497472 non-null  object 
 2   open    497461 non-null  float64
 3   high    497464 non-null  float64
 4   low     497464 non-null  float64
 5   close   497472 non-null  float64
 6   volume  497472 non-null  int64  
dtypes: float64(4), int64(1), object(2)
memory usage: 26.6+ MB


In [8]:
data.describe()

,open,high,low,close,volume
count,497461.000000,497464.000000,497464.000000,497472.000000,4.974720e+05
mean,86.352275,87.132562,85.552467,86.369082,4.253611e+06
std,101.471228,102.312062,100.570957,101.472407,8.232139e+06
min,1.620000,1.690000,1.500000,1.590000,0.000000e+00
25%,41.690000,42.090000,41.280000,41.703750,1.080166e+06
50%,64.970000,65.560000,64.353700,64.980000,2.084896e+06
75%,98.410000,99.230000,97.580000,98.420000,4.271928e+06
max,2044.000000,2067.990000,2035.110000,2049.000000,6.182376e+08


In [9]:
print(data.isnull().sum())

symbol     0
date       0
open      11
high       8
low        8
close      0
volume     0
dtype: int64


In [11]:
print("Total missing values:",data.isnull().sum().sum())

Total missing values: 27


In [12]:
numeric_cols=['open','high','low']
for col in numeric_cols:
    data[col]=data[col].fillna(data[col].median())

In [14]:
print(data.isnull().sum())
print("Total missing values:", data.isnull().sum().sum())

symbol    0
date      0
open      0
high      0
low       0
close     0
volume    0
dtype: int64
Total missing values: 0


In [15]:
data=data.dropna()

In [16]:
print(data.shape)

(497472, 7)


In [18]:
print('Duplicate rows:',data.duplicated().sum())

Duplicate rows: 0


In [19]:
data=data.drop_duplicates()
print(data.shape)

(497472, 7)


In [22]:
data['date'] = pd.to_datetime(data['date'])

In [23]:
print(data.dtypes)

symbol            object
date      datetime64[ns]
open             float64
high             float64
low              float64
close            float64
volume             int64
dtype: object


In [24]:
data['symbol'] = data['symbol'].str.upper().str.strip()

In [25]:
data.to_csv("Cleaned_Stock_Prices.csv", index=False)

In [26]:
# Check duplicate rows
print("Duplicate rows:", data.duplicated().sum())

# Remove duplicate rows
data = data.drop_duplicates()

# Verify new shape
print("Shape after removing duplicates:", data.shape)

Duplicate rows: 0
Shape after removing duplicates: (497472, 7)


In [27]:
print(data.dtypes)

symbol            object
date      datetime64[ns]
open             float64
high             float64
low              float64
close            float64
volume             int64
dtype: object


The dataset contained 27 missing values in the open, high, and low columns.

Missing values were imputed using the median of each respective column.

After imputation, the dataset contained no missing values.

Applying dropna() did not remove any rows because all missing values had already been handled.

The cleaned dataset retained all 497,472 rows and 7 columns.